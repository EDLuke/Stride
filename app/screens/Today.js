import React from 'react';
import {Font} from 'expo';
import { ScrollView, StyleSheet, View } from 'react-native';
import { Container, Content, List, Button, Icon, Text } from 'native-base';
import SingleFeed from '../components/layout/SingleFeed';
import { Octicons } from '@expo/vector-icons';

const renderFeed = (post, index) => (
  <SingleFeed
    key={index}
    name={post.name}
    username={post.username}
    content={post.content}
  />
)

export default class Today extends React.Component{
	state = {
    posts: [
      {
        profilePicture:'https://randomuser.me/api/portraits/thumb/men/83.jpg',
        name:'Romain',
        content:'I just hit my goal of the day!'
      },
      {
        profilePicture:'https://randomuser.me/api/portraits/thumb/women/45.jpg',
        name:'Mary',
        content:'I had cabbage, carrots and yogurt for lunch with about 1000 calorie'
      },
      {
        profilePicture:'https://randomuser.me/api/portraits/thumb/women/24.jpg',
        name:'Elizabeth',
        content:'I had steak for dinner with about 1200 calorie'
      },
      {
        profilePicture:'https://randomuser.me/api/portraits/thumb/women/39.jpg',
        name:'Shea',
        content:'I am looking for workout tips, any suggestions?'
      },
      {
        profilePicture:'https://randomuser.me/api/portraits/thumb/women/43.jpg',
        name:'Linda',
        content:'Just had a great workout burning 2000 calorie!'
      },
      {
        profilePicture:'https://randomuser.me/api/portraits/thumb/women/41.jpg',
        name:'Sindy',
        content:'Almost hit my goal today...will try harder tmr'
      },
      {
        profilePicture:'https://randomuser.me/api/portraits/thumb/men/21.jpg',
        name:'Jacob',
        content:'Finally at the top of my leaderboard!'
      }
    ]
	};

	render() {
		return (
			<Container>
       <Content>
          <List>
            {
              //console.log(this.state.posts)
              this.state.posts.map((feed, index) => renderFeed(feed, index))
            }
          </List>
        </Content>
        <Button
          rounded
          style={styles.button}
          onPress={() => Actions.newPost()}
        >
          <Octicons
            name="plus"
            size={16}
            color="#FFF"
            style=
            {{
            	left: 2,
            	padding: 15
            }}
          />
        </Button>
      </Container>
		);
	}
}

const styles = StyleSheet.create({
  container: {
    marginTop: 64,
  },
  button: {
    position: 'absolute',
    bottom: 20,
    right: 20,
  },
});