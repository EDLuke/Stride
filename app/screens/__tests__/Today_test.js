import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import Today from '../Today';
import { View } from 'react-native';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const wrapper = shallow(
      <Today/>
    );
  expect(wrapper).toMatchSnapshot();
});