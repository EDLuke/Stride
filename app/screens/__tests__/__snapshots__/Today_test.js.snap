// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <UnimplementedView
    style={
      Object {
        "backgroundColor": "#578CA9",
        "height": 56,
      }
    }
    title="Stride"
    titleColor="#FFF"
  />
  <Styled(Container)>
    <Styled(Content)>
      <Styled(List)>
        <Component
          content="I just hit my goal of the day!"
          key="0"
          name="Romain"
        />
        <Component
          content="I had cabbage, carrots and yogurt for lunch with about 1000 calorie"
          key="1"
          name="Mary"
        />
        <Component
          content="I had steak for dinner with about 1200 calorie"
          key="2"
          name="Elizabeth"
        />
        <Component
          content="I am looking for workout tips, any suggestions?"
          key="3"
          name="Shea"
        />
        <Component
          content="Just had a great workout burning 2000 calorie!"
          key="4"
          name="Linda"
        />
        <Component
          content="Almost hit my goal today...will try harder tmr"
          key="5"
          name="Sindy"
        />
        <Component
          content="Finally at the top of my leaderboard!"
          key="6"
          name="Jacob"
        />
      </Styled(List)>
    </Styled(Content)>
    <Styled(Button)
      onPress={[Function]}
      rounded={true}
      style={
        Object {
          "bottom": 20,
          "position": "absolute",
          "right": 20,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#FFF"
        name="plus"
        size={16}
        style={
          Object {
            "left": 2,
            "padding": 15,
          }
        }
      />
    </Styled(Button)>
  </Styled(Container)>
</View>
`;
